File: python/compress.py
def compress(s):
    s += '0'
    result = []
    i = 0
    j = 0
    while j < len(s):
        if s[j] != s[i]:
            count = j - i
            if count == 1:
                result.append(s[i])
            else:
              result.append(str(count) + s[i])
            i = j
        else:
            j += 1

    return ''.join(result)
            

File: ruby/compress.rb
def compress(s)
  s.concat('0')
  result = []
  i = 0
  j = 0
  while j < s.length do 
    if s[j] != s[i]
      count = j - i
      count == 1 ? result.append(s[i]) : result.append(count.to_s + s[i])
      i = j
    else 
      j += 1
    end
  end

  result.join('')
end